### PR TITLE
Extra apks ( useful for Orchestrator )

### PR DESCRIPTION
**Pull request by CristianGM**
depends on #28 
blocked by: https://github.com/gojuno/composer/pull/159

To run tests with Orchestrator two extra APKs are needed: test-services and orchestrator.
Once the PR to composer is hopefully merged the plugin will just provide those APKs to be installed.

Also I'll need to add to the docs that if you want to use Orchestrator with `clearPackageData` enabled you'll need to add: `instrumentationArgument("clearPackageData", "true")` unless we provide a better way ( like reading it directly from the androidExtension ).

What do you thing about it? Maybe I could just name it `androidTestUtilApks` because it's really doing all of them.

### CODE REVIEW

File: core/src/main/kotlin/com/trevjonez/composer/ComposerTask.kt [hunk 1]
```diff
@@ -77,10 +77,14 @@ open class ComposerTask : JavaExec(), ComposerConfigurator, ComposerTaskDsl {
       }
     }
 
+    val apks: List<File> = project.configurations.findByName("androidTestUtil")?.resolvedConfiguration?.files?.toList()
```
Comment (trevjonez) [inline core/src/main/kotlin/com/trevjonez/composer/ComposerTask.kt]: lets make it a lazy property similar to the apk and testApk.
then we can also move the default behavior of looking for androidTestUtil into the plugin since the core module doesn't directly depend on the AGP and androidTestUtil is an AGP thing.

I have been trying to keep core in the form of "bucket of vitamins anyone can use and customize" and plugin is "make it as easy as possible on the user"

File: core/src/main/kotlin/com/trevjonez/composer/ComposerParams.kt [hunk 2]
```diff
@@ -44,7 +45,8 @@ data class ComposerParams(
                 "--test-apk", testApk.absolutePath)
                 .let { params ->
                   withOrchestrator?.takeIf { it }?.let {
-                    params + arrayOf("--with-orchestrator")
+                    val apks = orchestratorApks.map { file -> file.absolutePath }.toTypedArray()
+                    params + arrayOf("--with-orchestrator", "--orchestrator-apks", *apks)
```
Comment (trevjonez) [inline core/src/main/kotlin/com/trevjonez/composer/ComposerParams.kt]: I wonder if it should just be "additional apk's" so that you could use it to facilitate something like "buy/install this other app to add features" in the end composer doesn't care what the extra apk is for right?

**Review by trevjonez — CHANGES_REQUESTED** [review]
for sure lets move the apks as a lazy property. but as for the additional apks vs orchestrator-apks it isn't up to me. just thought we could attempt for a more generally useful mechanism.

**Comment by CristianGM** [comment]
I should rebase it on master and fix an indentation change, but it won't happen today.

Let's see if it makes more sense to you now.

File: plugin/src/main/kotlin/com/trevjonez/composer/internal/ComposerBasePlugin.kt [hunk 3]
```diff
@@ -100,6 +103,11 @@ abstract class ComposerBasePlugin<T> : Plugin<Project>
                                    ?.takeIf { it.isPresent }
                                    ?.also { composerTask.dependsOn(it) }
                                ?: getOutputDir(composerTask))
+
+    composerTask.extraApks.setFrom(variantConfigurator?.extraApks
+        ?.takeUnless{ it.isEmpty }
+        ?.also { composerTask.dependsOn(it) }
+        ?: extraApks)
```
Comment (CristianGM) [inline plugin/src/main/kotlin/com/trevjonez/composer/internal/ComposerBasePlugin.kt:110]: I think I should simple do `comoosertTask.extraApks.setFrom(extraApks)`

I did it like this because I was trying to be nice, but it's nit possible to specify buildType o flavor on AndroidTestUtil

Comment (trevjonez) [reply plugin/src/main/kotlin/com/trevjonez/composer/internal/ComposerBasePlugin.kt:110]: I think I see what you mean but I need to look at it in idea. github is too hard to parse right now for some reason.

Comment (CristianGM) [reply plugin/src/main/kotlin/com/trevjonez/composer/internal/ComposerBasePlugin.kt:110]: It works right, maybe it's overkill, we can decide to keep it that way being consistent with others or...
make it simpler and ignore buildVariants for AndroidTestUtil and test if it works or not

**Review by trevjonez — APPROVED** [review]
